<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="statek_name" default="$(eval open(find('statek_config') + '/name').read())"/>
    <arg name="gamepad_addr" default="" />

    <!-- Run ds4 driver node -->
    <group ns="$(arg statek_name)/controller">
        <node pkg="ds4_driver" type="ds4_driver_node.py" name="ds4_driver" output="screen">
            <rosparam command="load" file="$(find ds4_driver)/config/params.yaml" />
            <param name="device_addr" type="string" value="$(arg gamepad_addr)" />
            <param name="use_standard_msgs" type="bool" value="false"/>
        </node>
    </group>

    <!-- Run teleop node -->
    <group ns="$(arg statek_name)/teleop">
        <node name="teleop_raw_ds4" pkg="statek_teleop" type="raw_ds4_node.py">
            <param name="statek_name" type="string" value="$(arg statek_name)" />
            <param name="gamepad_name" type="string" value="controller" />
            <rosparam command="load" file="$(find statek_config)/yaml/motion_config.yaml"/>
        </node>
    </group>
</launch>
